src/WebJobs.Script/Binding/WebJobsCoreScriptBindingProvider.cs

```diff
@@ -50,7 +64,20 @@ public override Type DefaultType
 
             public override Collection<Attribute> GetAttributes()
             {
-                var authLevel = Context.GetMetadataEnumValue<AuthorizationLevel>("authLevel", AuthorizationLevel.Function);
+                AuthorizationLevel defaultLevel = AuthorizationLevel.Function;
+                if (!string.IsNullOrEmpty(_parentRouteHandlingOptions?.AuthenticationLevel))
+                {
+                    try
+                    {
+                        defaultLevel = (AuthorizationLevel)Enum.Parse(typeof(AuthorizationLevel), _parentRouteHandlingOptions.AuthenticationLevel, ignoreCase: true);
```

Comment:
I recommend performing this parsing directly in `RouteHandlingOptions` during binding.